### PR TITLE
Fisheye

### PR DESCRIPTION
**Pull request by stephendb**
Adds 360 and Fisheye examples, sensor configs, and data frames.

### CODE REVIEW

File: examples/cpp/dev/camera_360.cpp [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+#include "sensor_config.h"
+#include "Stopwatch.h"
+
+#include "opencv2/core.hpp"
+#include "opencv2/core/utility.hpp"
+#include "opencv2/highgui.hpp"
+#include "opencv2/imgproc.hpp"
+
+// #define IMG_WIDTH 4096
+// #define IMG_HEIGHT 2160
```
Comment (bradleynull) [inline examples/cpp/dev/camera_360.cpp:21]: Remove these guys and just use one set of defines

File: examples/cpp/dev/camera_360.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        return -1;
+    }
+
+    // Configure the sensors we wish to use
+    std::vector<std::shared_ptr<Sensor>> sensors;
+
+    Camera360Config fc_config;
+    fc_config.server_ip = sim0.getServerIp();
+    fc_config.server_port = sim0.getServerPort();
+    fc_config.listen_port = 8100;// + i;
```
Comment (bradleynull) [inline examples/cpp/dev/camera_360.cpp:52]: Remove `+ i`

File: examples/cpp/dev/camera_360.cpp [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+#define IMG_HEIGHT 1024
+
+
+int main(int argc, char** argv)
+{
+    //Single Simulator Example
+    std::string server0_ip = "127.0.0.1";
+    int server_port = 8999;   // This has to be 8999 this simulator is listening for connections on this port;
+
+    //Read JSON files in cpp_client/config directory
```
Comment (bradleynull) [inline examples/cpp/dev/camera_360.cpp:33]: This is actually reading from the examples diretory.

File: examples/cpp/dev/camera_360.cpp [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    std::vector<std::shared_ptr<Sensor>> sensors;
+
+    Camera360Config fc_config;
+    fc_config.server_ip = sim0.getServerIp();
+    fc_config.server_port = sim0.getServerPort();
+    fc_config.listen_port = 8100;// + i;
+    fc_config.location.z = 225;
+    fc_config.rotation.pitch = -5;
+    fc_config.resolution = Resolution(IMG_WIDTH,IMG_HEIGHT);
+    fc_config.face_size = IMG_WIDTH/4;
```
Comment (bradleynull) [inline examples/cpp/dev/camera_360.cpp:56]: Maybe we should comment on why this is being divided by `4` here.

File: examples/cpp/dev/camera_360.cpp [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    }
+
+    //Get number of steps in scenario and start timer
+    mono::precise_stopwatch stopwatch;
+
+    /// initialize the vehicle, the first control command spawns the vehicle
+    sim0.sendControl(0, 0, 1, 1);
+
+    sensors[0]->sampleCallback = [](DataFrame *frame) {
+        // std::cout << "display" << std::endl;
```
Comment (bradleynull) [inline examples/cpp/dev/camera_360.cpp:80]: Remove.

File: examples/cpp/dev/camera_360.cpp [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    std::cout<<"***********ALL SENSOR's CONFIGS*******"<<std::endl;
+    for (auto& sensor : sensors)
+    {
+        sensor->configure();
+    }
+
+    //Get number of steps in scenario and start timer
+    mono::precise_stopwatch stopwatch;
+
+    /// initialize the vehicle, the first control command spawns the vehicle
```
Comment (bradleynull) [inline examples/cpp/dev/camera_360.cpp:76]: This isn't actually necessary to spawn the vehicle anymore.

File: examples/cpp/dev/camera_360.cpp [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+    // Send configurations to the simulator
+    std::cout<<"***********ALL SENSOR's CONFIGS*******"<<std::endl;
+    for (auto& sensor : sensors)
+    {
+        sensor->configure();
+    }
+
+    //Get number of steps in scenario and start timer
+    mono::precise_stopwatch stopwatch;
```
Comment (bradleynull) [inline examples/cpp/dev/camera_360.cpp:74]: This isn't being used as far as I can tell.

File: examples/cpp/dev/fisheye_camera.cpp [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+#define IMG_HEIGHT 1280
+
+
+int main(int argc, char** argv)
+{
+    //Single Simulator Example
+    std::string server0_ip = "127.0.0.1";
+    int server_port = 8999;   // This has to be 8999 this simulator is listening for connections on this port;
+
+    //Read JSON files in cpp_client/config directory
```
Comment (bradleynull) [inline examples/cpp/dev/fisheye_camera.cpp:33]: Same comment about examples directory.

File: examples/cpp/dev/fisheye_camera.cpp [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+        return -1;
+    }
+
+    // Configure the sensors we wish to use
+    std::vector<std::shared_ptr<Sensor>> sensors;
+
+    FisheyeCameraConfig fc_config;
+    fc_config.server_ip = sim0.getServerIp();
+    fc_config.server_port = sim0.getServerPort();
+    fc_config.listen_port = 8100;// + i;
```
Comment (bradleynull) [inline examples/cpp/dev/fisheye_camera.cpp:52]: Remove `+ i`.

File: examples/cpp/dev/fisheye_camera.cpp [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    fc_config.location.z = 225;
+    fc_config.resolution = Resolution(IMG_WIDTH,IMG_WIDTH);
+    fc_config.fov = 180.f;
+    fc_config.rotation.yaw = -90.f;
+    fc_config.face_size = 1024;
```
Comment (bradleynull) [inline examples/cpp/dev/fisheye_camera.cpp:57]: Might want to comment on recommended values for `face_size` here.

File: examples/cpp/dev/fisheye_camera.cpp [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    std::cout<<"***********ALL SENSOR's CONFIGS*******"<<std::endl;
+    for (auto& sensor : sensors)
+    {
+        sensor->configure();
+    }
+
+    //Get number of steps in scenario and start timer
+    mono::precise_stopwatch stopwatch;
+
+    /// initialize the vehicle, the first control command spawns the vehicle
```
Comment (bradleynull) [inline examples/cpp/dev/fisheye_camera.cpp:77]: Same comment. Isn't needed for spawning.

File: examples/cpp/dev/fisheye_camera.cpp [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+
+    // Send configurations to the simulator
+    std::cout<<"***********ALL SENSOR's CONFIGS*******"<<std::endl;
+    for (auto& sensor : sensors)
+    {
+        sensor->configure();
+    }
+
+    //Get number of steps in scenario and start timer
+    mono::precise_stopwatch stopwatch;
```
Comment (bradleynull) [inline examples/cpp/dev/fisheye_camera.cpp:75]: Not used?

File: examples/cpp/dev/camera_360.cpp (hunk 1 above)
Comment (stephendb) [reply examples/cpp/dev/camera_360.cpp:21]: fixed

File: examples/cpp/dev/camera_360.cpp (hunk 2 above)
Comment (stephendb) [reply examples/cpp/dev/camera_360.cpp]: fixed

File: examples/cpp/dev/camera_360.cpp (hunk 3 above)
Comment (stephendb) [reply examples/cpp/dev/camera_360.cpp]: fixed

Comment (stephendb) [reply examples/cpp/dev/camera_360.cpp]: Also, I copy pasted from Camera for these examples, so lots of these errors are probably across our examples.

File: examples/cpp/dev/camera_360.cpp (hunk 4 above)
Comment (stephendb) [reply examples/cpp/dev/camera_360.cpp]: added

File: examples/cpp/dev/camera_360.cpp (hunk 5 above)
Comment (stephendb) [reply examples/cpp/dev/camera_360.cpp]: fixed

File: examples/cpp/dev/camera_360.cpp (hunk 6 above)
Comment (stephendb) [reply examples/cpp/dev/camera_360.cpp]: fixed

File: examples/cpp/dev/camera_360.cpp (hunk 7 above)
Comment (stephendb) [reply examples/cpp/dev/camera_360.cpp]: fixed

File: examples/cpp/dev/fisheye_camera.cpp (hunk 8 above)
Comment (stephendb) [reply examples/cpp/dev/fisheye_camera.cpp]: fixed

File: examples/cpp/dev/fisheye_camera.cpp (hunk 9 above)
Comment (stephendb) [reply examples/cpp/dev/fisheye_camera.cpp]: fixed

File: examples/cpp/dev/fisheye_camera.cpp (hunk 10 above)
Comment (stephendb) [reply examples/cpp/dev/fisheye_camera.cpp:53]: fixed

File: examples/cpp/dev/fisheye_camera.cpp (hunk 11 above)
Comment (stephendb) [reply examples/cpp/dev/fisheye_camera.cpp]: fixed

File: examples/cpp/dev/fisheye_camera.cpp (hunk 12 above)
Comment (stephendb) [reply examples/cpp/dev/fisheye_camera.cpp]: fixed

**Pull request merged**